boards/apollo3/lora_things_plus/src/main.rs

```diff
@@ -113,6 +116,13 @@ static mut BME280: Option<
 static mut CCS811: Option<&'static capsules_extra::ccs811::Ccs811<'static>> = None;
 #[cfg(feature = "atecc508a")]
 static mut ATECC508A: Option<&'static capsules_extra::atecc508a::Atecc508a<'static>> = None;
+#[cfg(feature = "chirp_i2c_moisture")]
+static mut CHIRP_I2C_MOISTURE: Option<
```

Comment:
So I know that this is following the established pattern, but is actually really problematic in practice now that use of static muts is a warning (and will be promoted to a hard error in the next Rust edition). I don't know whether it's worth blocking this PR any further on, but we will have to go through all of these tests and remove those static usages eventually. Right now this only passes our CI because we don't check for warnings when we run tests (which is bad). I devised a fix for this that I should upstream at some point, but it'll obviously make all of these tests fail.

I think we'll instead want reusable setup functions (like this board has anyways),, that a `#[test]` function then calls to instantiate the peripherals and pass `&'static mut` references to the functions that actually execute the tests. I had a much better grasp on this a while ago, I'll see whether I can dig this back up.

Reply:
I think that all of the tests in a range of boards will need to be updated to not use static muts and this PR shouldn't block on that. We can merge this and then still update all of the static muts in the future

Reply:
We discussed this on the core call today. I think that you make a fair argument by saying that this follows an established paradigm.

Ultimately, this is a very unfortunate situation. A while back we accidentally started permitting warnings in the CI tests. Then, by updating the Rust toolchain, we introduced these `static mut` warnings but did not notice how prevalent this pattern was in the integration tests. Now, we're faced with the prospect of these warnings **becoming hard errors in the 2024 Rust edition**, which will be released imminently.

I strongly believe that it's not solely on you to fix these errors. Furthermore, for these tests, we don't want to simply fix them by using raw pointer dereferences -- the hack that we used to get rid of these warnings on master was really just that, a hack, and I'd rather not have more of those in the codebase stick around forever. Instead, we'll want to refactor those tests to instead pass such instances as parameters and avoid going via global variables entirely.

However, I don't think that adding _more_ of these tests is a good idea either. To avoid blocking this PR, can we suggest to split out the tests (which already confirm that this code works!) into a different PR? This would avoid introducing more mutable statics into our codebase, and we can also use that PR to see how we can convert these existing tests into ones that don't use mutable statics.

Reply:
Done!